File: server/src/ChinaImportPlatform.Api/Models/Announcement.cs
namespace ChinaImportPlatform.Api.Models;

/// <summary>A one-to-many broadcast from the seller, shown in the feed and pushed.</summary>
public class Announcement : BaseEntity
{
    public string Title { get; set; } = string.Empty;

    public string Body { get; set; } = string.Empty;

    public string? ImageS3Key { get; set; }

    public DateTime? CutoffDate { get; set; }

    /// <summary>Audience segment, e.g. "all" or a category name. Null means all customers.</summary>
    public string? Segment { get; set; }

    public DateTime? PublishedAt { get; set; }
}

File: client/china_import_app/lib/features/announcements/domain/entities/announcement.dart
/// Domain representation of a platform announcement.
class Announcement {
  const Announcement({
    required this.id,
    required this.title,
    required this.body,
    required this.publishedAt,
    this.imageUrl,
    this.isPinned = false,
  });

  final String id;
  final String title;
  final String body;
  final DateTime publishedAt;
  final String? imageUrl;
  final bool isPinned;
}
